garnett/trajectory.py

```diff
@@ -352,7 +341,7 @@ def make_snapshot(self):
     def copyto_snapshot(self, snapshot):
```

Comment:
> In this PR, the function `copy_to_hoomd_blue_snapshot` is renamed to `to_hoomd_snapshot`. However, the `Frame` and `FrameData` classes still have methods named `copyto_snapshot`. I propose the following changes:
> 
> * The **module-level function** `to_hoomd_snapshot` should be made into a private module-level function `_to_hoomd_snapshot`. The only reason the module-level function exists here is to provide a method that can be used by both `Frame` and `FrameData` without introducing an inheritance relationship between the two classes.
> * The deprecated **module-level function** `copy_to_hoomd_blue_snapshot` message should be changed to "This function will be **removed** in 0.8.0" and the implementation should be switched from calling `to_hoomd_snapshot` to `_to_hoomd_snapshot`.
> * The **class methods** `copyto_snapshot` of the `Frame` and `FrameData` classes should be renamed to `to_hoomd_snapshot`.
> * The **class methods** `copyto_snapshot` of the `Frame` and `FrameData` classes should raise deprecation warnings and internally call the class method `self.to_hoomd_snapshot`.
> * The **class methods** `make_snapshot` of the `Frame` and `FrameData` classes should be deprecated and removed in 0.8.0. The signature of `to_hoomd_snapshot` should have `snapshot=None` by default, which would return a snapshot object (as was previously done by `make_snapshot`.
> * It does not appear that the **module-level function** `from_hoomd_snapshot` is used or tested anywhere in the package. I would consider making it a private method `_from_hoomd_snapshot`. Similarly, `to_hoomd_snapshot` does not appear to be tested.
> * The deprecated **module-level function** `copyfrom_hoomd_blue_snapshot` message should be changed to "This function will be **removed** in 0.8.0" and the implementation should be switched from calling `from_hoomd_snapshot` to `_from_hoomd_snapshot`.
> 
> I am also unsure whether the `FrameData` class should remain a public class, or whether the `Frame.frame_data` attribute should remain public. I am uncertain what the rationale was for this design. From my experience with the garnett API, it would be sensible to make `FrameData` private as `_FrameData` and to make `Frame.frame_data` into `Frame._frame_data`. In that case, I would also remove the `to_hoomd_snapshot` and other user-facing methods in the `_FrameData` class, which would be purely an internal storage container. Notably, `len(FrameData)` returns the number of types while `len(Frame)` returns the number of particles. Longer-term, it may be a good strategy to unify the `_RawFrameData` and `_FrameData` classes -- I can't see the reason for this class structure.
> 

I have wondered this  myself before and found it quite confusing. I think it is a good idea to review the implementation and unify the classes.

> **I would recommend holding these changes for a follow-up pull request** before the next minor release (since we're aiming to deprecate and break all needed APIs in the same minor release).

Reply:
Just to clarify, does this mean we should revert canges to `copyto_hoomd_blue_snapshot` and `copyfrom_hoomd_blue_snapshot`, then file a seperate issue? Or to leave current changes?

Reply:
@klywang You can leave the changes in this PR. Creating an issue describing those changes would be great!

Reply:
@bdice Just want to make sure I understood correctly: leave the changes to the module level functions we've made thus far here, and create a separate PR for the the rest of the changes?

Reply:
@lyrivera Yeah, I think so. We should get this PR finished/merged. Thanks!

Reply:
@bdice Yes, let's have it merged by the end of the week. @klywang @gaofy95 Whats left?

Reply:
Reverted the function names back. There was something about testing the stacklevel of the deprecation warning. We added tests for deprectation but I can't find any documentation about checking stack level in `unittest`. I think we covered everything else.

Reply:
@klywang  I think the original names of the functions were `copyto_hoomd_blue_snapshot` and `copyfrom_hoomd_blue_snapshot` (without the first underscore)

Reply:
I reverted the names to the original ones. I'm preparing a separate PR to fix this and address #138.